src/arena.rs

```diff
@@ -99,7 +102,7 @@ impl<T: Send + Sync, const BITARRAY_LEN: usize, const LEN: usize> Arena<T, BITAR
     ///
     /// This function is lock-free.
     pub fn try_insert(&self, mut value: T) -> Result<ArenaArc<T, BITARRAY_LEN, LEN>, (T, u32)> {
-        let slice = self.buckets.as_slice();
+        let slice = &*self.buckets.as_slice();
```

Comment:
I don't think we need this reborrow, it does not change the code by any mean:

```suggestion
        let slice = self.buckets.as_slice();
```